### DISCUSSION TITLE
Alternative to templates for x-for?

### DISCUSSION THREAD

**Discussion by querkmachine** [alpinejs/alpine#2633] [1. Help]
I have a situation where I'm constructing a list of `<option>`s inside of a `<select>` using nested x-for and x-if statements. 

Unfortunately, the requirements that x-for can only be on `<template>` elements, and that `<template>` elements must always have a single root element, makes constructing semantically valid HTML seemingly impossible. 

Example: https://jsfiddle.net/querkmachine/y2x6kzp8/

In this example, only the first nested `<template>` element is evaluated, the second one is ignored. The same happens if the `<template>`s are removed and the x-if attributes moved to their child elements. In both cases this issue is related to the requirement that a template have only a single child element.

I can work around it by having two separate loops, but this means I can't mix and match grouped and standalone options. Presumably I could also resolve this by constructing the resulting HTML inside of a method, rather than using template functions (but that seems to defy the point somewhat).

I'm still relatively new to using Alpine, though I'm familiar with Vue. Is there an approach to this that is compatible with how Alpine does things, or is what I'm after just not possible?

**Comment by solution-loisir** [comment]
Hi @querkmachine, I think that a work around would be to wrap all the nested content in a `<div>` which would act as the single root element for your loop. I agree that the single root element can be restrictive. The `<template>` element doesn't have that limitation in the wild.

**Reply by querkmachine** [reply]
> Hi @querkmachine, I think that a work around would be to wrap all the nested content in a `<div>` which would act as the single root element for your loop. 

That's not possible in this situation. As I'm working within a `<select>` element, the only elements that are allowed are `<option>`, `<optgroup>` and `<template>`. `<template>` can't be used as the root element, and using the other two automatically creates a difference in semantics or appearance, thus the conundrum.

I ended up just constructing the dropdown inner HTML inside of a method (deadlines, y'know), but it'd be nice if it were possible to do otherwise in a future version of Alpine.

**Reply by solution-loisir** [reply]
Thanks for the follow up! I guess Alpine would need something like React `<Fragment/>`.
